packages/ast-spec/tsconfig.build.json

```diff
@@ -1,10 +1,9 @@
 {
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "composite": true,
-    "jsx": "preserve",
-    "outDir": "./dist",
-    "rootDir": "./src",
+    "rootDir": "src",
+    "outDir": "dist",
+    "emitDeclarationOnly": false,
```

Comment:
Oh, hm, you're using `emitDeclarationOnly`, but only sometimes. It seems simpler to only set noEmit for test projects, then leave all emit on in the base config, then only have `tsconfig.json` and `tsconfig.spec.json` per-package if you want to split apart tests.

Reply:
Let me know if my comment here https://github.com/typescript-eslint/typescript-eslint/pull/9038#discussion_r1608042136 alters your view on this please. Having type-checking not produce .js etc artifacts would seem to be desirable, and therefore we would need a way to distinguish between type-checking vs building